types/rdf-dataset-indexed/dataset.d.ts

```diff
@@ -0,0 +1,39 @@
+import * as RDF from 'rdf-js';
+import { Readable } from 'stream';
+
+declare class DatasetIndexed implements RDF.Dataset<RDF.Quad> {
```

Comment:
Could be `Q extends RDF.Quad = RDF.Quad` and the factory/some of the methods updated?